File: API_Polideportivo/app/schemas/reserva.py
from pydantic import BaseModel
from datetime import datetime

class ReservaCreate(BaseModel):
    pista_id: int
    fecha_hora_inicio: datetime
    fecha_hora_fin: datetime

class ReservaOut(BaseModel):
    id: int
    pista_id: int
    usuario_id: int
    fecha_hora_inicio: datetime
    fecha_hora_fin: datetime
    nombre_pista: str
    precio_total: float

    class Config:
        orm_mode = True

File: WebPolideportivo/WebPolideportivo/Models/Reserva.cs
using System.Text.Json.Serialization;

namespace WebPolideportivo.Models;

public class Reserva
{
    public int Id { get; set; }
    public int PistaId { get; set; }
    public int UsuarioId { get; set; }

    [JsonPropertyName("fecha_hora_inicio")]
    public DateTime FechaHoraInicio { get; set; }
    [JsonPropertyName("fecha_hora_fin")]
    public DateTime FechaHoraFin { get; set; }

    [JsonPropertyName("nombre_pista")]
    public string NombrePista { get; set; } = string.Empty;
    [JsonPropertyName("precio_total")]
    public decimal PrecioTotal { get; set; }

    public DateTime Fecha => FechaHoraInicio.Date;
    public string HoraInicio => FechaHoraInicio.ToString("HH:mm");
    public string HoraFin => FechaHoraFin.ToString("HH:mm");
}



